### DISCUSSION TITLE
Problem with $webevent.data

### DISCUSSION THREAD

**Discussion by IrbisDiamond88** [Badgerati/Pode#1122] [Q&A]
Hello. 
Colleagues, I ask for your help.
I've been struggling with this problem for several days now. In the $webevent.data hash table, fragments of the request request appear in the key of one of the elements. 
More specifically, it looks like this:
` [ Content-Length: 71 key, value ] [key, value] [key, value]`
As you understand, it is not possible to use the value from the table, since the order of the keys in the table is absolutely unpredictable.

**Comment by behrouzamiri** [comment]
> Hello. 
> Colleagues, I ask for your help.
> I've been struggling with this problem for several days now. In the $webevent.data hash table, fragments of the request request appear in the key of one of the elements. 
> More specifically, it looks like this:
> ` [ Content-Length: 71 key, value ] [key, value] [key, value]`
> As you understand, it is not possible to use the value from the table since the order of the keys in the table is absolutely unpredictable.

I'm not sure if i got this right, but You can get the value simply using
```$WebEvent.Data["yourkeyname"]```

The point is, you probably have a rout with the `Post` method, and you usually should allow certain specific data to be posted, meaning that you should know and expect the exact key name.

If you're not sure what the data that is being posted looks like (e.g., external system that you don't control and don't have documentation for it), you can simply receive the data and save it, then you can easily play with that later.
Like in your route:
```$WebEvent | Export-CliXml -path <somepath\webevent.xml>```
Then, after a successful post request, you'll have the file webevent.xml in the path. Then, load it into a variable like:
```$WebEvent = Import-CliXml -path <somepath\webevent.xml>```
